### PR TITLE
feat(file-upload): file-upload twig implementation - TWIG-30

### PR DESCRIPTION
**Pull request by serotonine**
# PR description

Please drop a few lines about the PR: what it does, how to test it, etc.

## QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

- [ ] I have put the vanilla component as `devDependencies`
- [ ] I have put the specs package as `devDependencies`
- [ ] I have added the components directly used in the twig file (with `include` or `embed`) as `dependencies`
- [ ] My component is listed in `@ecl-twig/ec-components`'s `dependencies`
- [ ] My variables naming follow the guidelines (snake case for twig)
- [ ] I have provided tests
- [ ] I have provided documentation (for the "notes" tab)
- [ ] If my local `yarn.lock` contains changes, I have committed it
- [ ] I have given my PR the proper label (`pr: review needed` to indicate that I'm done and now waiting for a review ,`pr: wip` to indicate that I'm actively working on it ...)

### CODE REVIEW

File: src/ec/packages/ec-component-file-upload/file-upload.story.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+storiesOf('Components/Forms/File Upload', module)
+  .addDecorator(withKnobs)
+  .addDecorator(withNotes)
+  .addDecorator(withCode)
+  .add(
+    'default',
+    () =>
+      fileUpload({
+        ...dataDefault,
```
Comment (planctus) [inline src/ec/packages/ec-component-file-upload/file-upload.story.js]: Not sure you really need the ...dataDefault here for the way you are setting the story, all the properties are redefined later.

File: src/ec/packages/ec-component-file-upload/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# ECL Twig - EC File Upload component
```
[the remark this answers is not in the document]
Comment (serotonine) [reply src/ec/packages/ec-component-file-upload/README.md]: Ok.  I use only the README.md file

File: src/ec/packages/ec-component-file-upload/file-upload.html.twig [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
+  {% endfor %}
+{% endif %}
+
+{# Print the result #}
+
+<div class="{{ _css_class }}"{{ _extra_attributes|raw }}>
+    
+    {# label #}
+    {% if _label is not empty %}
```
[the remark this answers is not in the document]
Comment (serotonine) [reply src/ec/packages/ec-component-file-upload/file-upload.html.twig]: ok

File: src/ec/packages/ec-component-file-upload/file-upload.story.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        label: text('Label', dataDefault.label),
+        helperText: text('Help message', dataDefault.helperText),
+        invalid: boolean('Invalid', false),
+        disabled: boolean('Disabled', false),
+        required: boolean('Required', false),
+        invalidText: text('invalid text', dataDefault.invalidText),
+        requiredText: text('Required text', dataDefault.requiredText),
+        optionalText: text('Optional Text', dataDefault.optionalText),
+        extra_classes: text('Extra classes', ''),
+        extra_attributes: object('Extra attributes', [
```
[the remark this answers is not in the document]
Comment (serotonine) [reply src/ec/packages/ec-component-file-upload/file-upload.story.js]: ok

File: src/ec/packages/ec-component-file-upload/package.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "@ecl-twig/ec-component-file-upload",
+  "author": "European Commission",
+  "license": "EUPL-1.1",
+  "version": "2.9.0",
```
[the remark this answers is not in the document]
Comment (serotonine) [reply src/ec/packages/ec-component-file-upload/package.json]: ok

File: src/ec/packages/ec-component-file-upload/package.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+   "dependencies": {
+    "@ecl/ec-component-form-feedback-message": "^2.9.0",
+    "@ecl/ec-component-form-form-group": "^2.9.0",
+    "@ecl/ec-component-form-help-block": "^2.9.0",
+    "@ecl/ec-component-form-label": "^2.9.0",
+    "@ecl/ec-specs-file-upload": "^2.9.0"
+  },
+  "devDependencies": {
+    "@ecl/ec-component-form-file-upload": "^2.9.0"
+
```
[the remark this answers is not in the document]
Comment (serotonine) [reply src/ec/packages/ec-component-file-upload/package.json]: ok

File: src/ec/packages/ec-component-file-upload/file-upload.html.twig [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    
+    {#invalid text#}
+    {% if _invalid and _invalid_text is not empty %}
+        <div class="ecl-feedback-message">
+            {%- block invalid_text _invalid_text -%}
+        </div>
+    {% endif %}
+    
+    {#input#}
+    <input
```
[the remark this answers is not in the document]
Comment (serotonine) [reply src/ec/packages/ec-component-file-upload/file-upload.html.twig]: yes I think it's more readable to separate elements.

File: src/ec/packages/ec-component-file-upload/file-upload.html.twig [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+{% endif %}
+
+{# Print the result #}
+
+<div class="{{ _css_class }}"{{ _extra_attributes|raw }}>
+    
+    {# label #}
+    {% if _label is not empty %}
+    {% set _class_prefix = 'ecl-form-label' %}
+        <label class="ecl-form-label ecl-form-label--invalid" for="example">
```
[the remark this answers is not in the document]
Comment (serotonine) [reply src/ec/packages/ec-component-file-upload/file-upload.html.twig]: ok. the for I forgot to set the variable (_input_id). I fix the --invalid modifer

File: src/ec/packages/ec-component-file-upload/file-upload.html.twig [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+    {% if _invalid and _invalid_text is not empty %}
+        <div class="ecl-feedback-message">
+            {%- block invalid_text _invalid_text -%}
+        </div>
+    {% endif %}
+    
+    {#input#}
+    <input 
+        type="file" 
+        id="{{-_input_id -}}"
```
[the remark this answers is not in the document]
Comment (serotonine) [reply src/ec/packages/ec-component-file-upload/file-upload.html.twig]: ok

File: src/ec/packages/ec-component-file-upload/file-upload.test.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+      const withExtraAttributes = merge(dataDefault, {
+        extra_attributes: [
+          { name: 'data-test', value: 'data-test-value' },
+          { name: 'data-test-1', value: 'data-test-value-1' },
+        ],
+      });
+
+      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
+    });
```
[the remark this answers is not in the document]
Comment (serotonine) [reply src/ec/packages/ec-component-file-upload/file-upload.test.js]: ok

File: src/ec/packages/ec-component-file-upload/file-upload.story.js (hunk 1 above)
Comment (serotonine) [reply src/ec/packages/ec-component-file-upload/file-upload.story.js]: ok

File: src/ec/packages/ec-component-file-upload/__snapshots__/file-upload.test.js.snap [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+    class="ecl-file-upload"
+    required=""
+    type="file"
+  />
+</div>
+`;
+
+exports[`EC - File Upload  Default renders correctly with extra attributes 1`] = `
+<div
+  class="ecl-form-group ecl-form-group--file-upload"
```
Comment (emeryro) [inline src/ec/packages/ec-component-file-upload/__snapshots__/file-upload.test.js.snap]: no extra attributes have been added here

File: src/ec/packages/ec-component-file-upload/file-upload.story.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+        disabled: boolean('Disabled', false),
+        invalid: boolean('Invalid', false),
+        id: text('Input id', dataDefault.id),
+        name: text('Name', dataDefault.name),
+        label: text('Label', dataDefault.label),
+        helper_text: text('Help message', dataDefault.helper_text),
+        invalid_text: text('Invalid text', 'This is an error message'),
+        required_text: text('Required text', '*'),
+        optional_text: text('Optional Text', '(optional)'),
+        extra_classes: text('Extra classes', ''),
```
Comment (emeryro) [inline src/ec/packages/ec-component-file-upload/file-upload.story.js]: There is no need to expose extra classes here

**Pull request merged**